### PR TITLE
Fix #7391, Starting from live map widget crashes the app

### LINKED ISSUE #7303: NPE on  map
New crash report with 2019.02.17-RC. 
Not sure if this can be related to sdk-raise or not.

Device: Android 9.0

Crash report:
```
java.lang.NullPointerException: 
 
  at io.reactivex.internal.functions.ObjectHelper.requireNonNull (ObjectHelper.java:39)
 
  at io.reactivex.Observable.combineLatest (Observable.java:433)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:111)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:77)
 
  at cgeo.geocaching.maps.mapsforge.v6.NewMap.initializeLayers (NewMap.java:784)
 
  at cgeo.geocaching.maps.mapsforge.v6.NewMap.onStart (NewMap.java:725)
 
  at android.app.Instrumentation.callActivityOnStart (Instrumentation.java:1392)
 
  at android.app.Activity.performStart (Activity.java:7157)
 
  at android.app.ActivityThread.handleStartActivity (ActivityThread.java:2952)
 
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:180)
 
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:165)
 
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:142)
 
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:70)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1823)
 
  at android.os.Handler.dispatchMessage (Handler.java:107)
 
  at android.os.Looper.loop (Looper.java:198)
 
  at android.app.ActivityThread.main (ActivityThread.java:6729)
 
  at java.lang.reflect.Method.invoke (Native Method)
 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:858)
```

### LINKED ISSUE #7391: Starting from live map widget crashes the app
##### Detailed steps causing the problem:
- Create a new widget on your device "desktop" with the function "live map"
- Make sure c:geo is not running in background (exit completely / hard close)
- Click the live map widget

##### Actual behavior after performing these steps:
Immediate crash

##### Expected behavior after performing these steps:
Live map shall be started.
(It works only if c:geo was active before and is still in background tasks)

##### Version of c:geo used:
2019.02.23

##### Is the problem reproducible for you?
Yes

##### System information:
Samsung S7, Android 8.0

### PR DESCRIPTION
**Pull request by rsudev**
- check location permission and start location provider even on map
- possibly fixes #7303 and #7300 as well

### CODE REVIEW

File: main/src/cgeo/geocaching/maps/CGeoMap.java [hunk 1]
```diff
@@ -525,7 +528,20 @@ private void prepareFilterBar() {
...
     public void onResume() {
         super.onResume();
-        resumeDisposables.addAll(geoDirUpdate.start(GeoDirHandler.UPDATE_GEODIR), startTimer());
+
+        // resume location access
+        PermissionHandler.executeIfLocationPermissionGranted(this.activity, new PermissionGrantedCallback(PermissionRequestContext.CacheDetailActivity) {
+
+            @Override
+            public void execute() {
+                Log.d("CacheDetailActivity.OnResumePermissionGrantedCallback.execute");
```
Comment (Bananeweizen) [inline main/src/cgeo/geocaching/maps/CGeoMap.java]: copy paste? CacheDetailActivity?

File: main/src/cgeo/geocaching/maps/mapsforge/v6/NewMap.java [hunk 2]
```diff
@@ -781,7 +784,19 @@ private void initializeLayers() {
             targetView.setTarget(target.getGeocode(), target.getName());
         }
 
-        this.resumeDisposables.add(this.geoDirUpdate.start(GeoDirHandler.UPDATE_GEODIR));
+        // resume location access
+        PermissionHandler.executeIfLocationPermissionGranted(this, new PermissionGrantedCallback(PermissionRequestContext.CacheDetailActivity) {
+
+            @Override
+            public void execute() {
+                Log.d("CacheDetailActivity.OnResumePermissionGrantedCallback.execute");
```
Comment (Bananeweizen) [inline main/src/cgeo/geocaching/maps/mapsforge/v6/NewMap.java]: same here

File: main/src/cgeo/geocaching/maps/CGeoMap.java (hunk 1 above)
Comment (rsudev) [reply main/src/cgeo/geocaching/maps/CGeoMap.java]: Thanks for catching. Yes, copy paste from CacheDetailActivity. I was thinking how to encapsulate that, as most of the code is the same (only line 542 is specific for this location.

**Comment by Lineflyer** [comment]
@Bananeweizen: OK for you now?
Would like to test drive this change on a new beta version in the field to see if it also fixes #7303

**Comment by Lineflyer** [comment]
As the comments by @Bananeweizen have been fixed by @rsudev I assume it to be ok now.

**Pull request merged**